cfme/common/provider.py

```diff
@@ -404,12 +426,20 @@ def _do_stats_match(self, client, stats_to_match=None, refresh_timer=None, ui=Fa
 
     @property
     def exists(self):
-        ems = cfmedb()['ext_management_systems']
-        provs = (prov[0] for prov in cfmedb().session.query(ems.name))
-        if self.name in provs:
+        """ Checks if exists among managed providers
+
+        Note:
+            If provider exists but with a different name, name is updated
+        """
+        for prov in store.current_appliance.managed_providers:
```

Comment:
So, my point was that looking at ```store.current_appliance.managed_providers``` I can't tell if it is REST/UI whatever, AND it could change without us knowing. So when we move exists to sentaku methods and have defined implementations, we potentially want to define the method a little more explicitly.